tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals.bal

```diff
@@ -402,3 +402,17 @@ function testQueryInXMLTemplateExpr() {
     // var x8 = xml `<doc>${xml `foo` + (from string s in ["a", "b"] select xml `${s}z`)}</doc>`; // issue #36541
     // test:assertEquals(x8.toString(), "<doc>fooazbz</doc>")
 }
+
+function testXMLLiteralWithConditionExpr() {
+    xml? s = xml `foo`;
+    string target = "foo";
+
+    xml x1 = (s ?: xml `<baz/>`);
+    test:assertEquals(x1.toString(), target);
+
+    xml x2 = (s is xml ? s : xml `<baz/>`) + xml ``;
+    test:assertEquals(x2.toString(), target);
+
+    xml x3 = (s ?: xml `<baz/>`) + xml ``;
+    test:assertEquals(x3.toString(), target);
+}
```

Comment:
Shall we add some more tests with subtypes of `xml`?
Ex:
```ballerina
type XMLType xml:Comment|xml:ProcessingInstruction;

public function main() {
    xml? s = ();
    xml:Element e1 = xml `<baz/>`; 
    xml v1 = (s ?: e1) + xml `<a>Some Element</a>`; 

    xml<xml:Element> e2 = xml `<baz/>`;
    xml v2 = (s ?: e2) + xml `<a>Some Element</a>`; 

    XMLType e3 = xml `<!--This is a comment-->`;
    xml v3 = (s ?: e3) + xml `<a>Some Element</a>`;
}
```

Reply:
Updated 486b106f2c018bd3a60e5d5ac5c774bd5ab8e3df